soroban-env-host/src/native_contract/token/balance.rs

```diff
@@ -44,9 +44,8 @@ fn write_balance(e: &Host, id: Identifier, amount: i128) -> Result<(), HostError
 
 // Metering: covered by components.
 pub fn receive_balance(e: &Host, id: Identifier, amount: i128) -> Result<(), HostError> {
-    let is_frozen = read_state(e, id.clone())?;
-    if is_frozen {
-        return Err(e.err_status_msg(ContractError::BalanceFrozenError, "balance is frozen"));
+    if is_deauthorized(e, id.clone())? {
+        return Err(e.err_status_msg(ContractError::BalanceFrozenError, "balance is deauthorized"));
```

Comment:
Shall we also rename the error?

Reply:
Ah I had `BalanceDeauthorizedError` but reverted it when I decided to keep the freeze/unfreeze terminology. Looks like I didn't update the message though. If we decide to keep freeze/unfreeze for now, I'll update the message back to `"balance is frozen"`.